Packs/CrowdStrikeFalconStreamingV2/Integrations/CrowdStrikeFalconStreamingV2/CrowdStrikeFalconStreamingV2.py

```diff
@@ -55,17 +56,27 @@ def set_auth_headers(self, token: str) -> None:
         self._headers.update({'Authorization': f'Bearer {token}'})
         demisto.debug('Set auth headers successfully')
 
-    def discover_stream(self) -> Dict:
+    async def discover_stream(self, refresh_token: 'RefreshToken') -> Dict:
         demisto.debug('Sending request to discover stream')
-        return self._http_request(
-            method='GET',
-            url_suffix='/sensors/entities/datafeed/v2',
-            params={'appId': self.app_id},
-        )
-
-    def refresh_stream_session(self) -> None:
+        while True:
+            try:
+                return self._http_request(
+                    method='GET',
+                    url_suffix='/sensors/entities/datafeed/v2',
+                    params={'appId': self.app_id},
+                )
+            except DemistoException as e:
+                if 'Error in API call [401]' in str(e):
+                    sleep_time = uniform(1, 10)
+                    demisto.debug(f'Got status code 401 on stream discovery, going to sleep for {sleep_time} - {str(e)}')
```

Comment:
is there any chance we are writing to logs `demisto.debug` in parallel?

Reply:
didn't encounter such issue in that integration in any env before